### PR TITLE
Merge master into uar

### PR DESCRIPTION
**Pull request by SteveJacksonSoft**
Created a temporary branch based on master and merged uar-test into it.

Now merging it into uar-test.

Had to resolve merge conflicts in 3 files:
* Jenkinsfile
* DelegateUserDataService.cs
* DelegateUploadFileServiceTests.cs

### CODE REVIEW

File: DigitalLearningSolutions.Data.Tests/Services/DelegateUploadFileServiceTests.cs [hunk 1]
```diff
@@ -258,7 +258,7 @@ public void ProcessDelegateTable_has_missing_PRN_error_for_HasPRN_true_with_miss
         [Test]
         public void ProcessDelegateTable_has_invalid_PRN_characters_error_for_PRN_with_invalid_characters()
         {
-            var row = GetSampleDelegateDataRow(hasPrn: true, prn: "^%�PRN");
+            var row = GetSampleDelegateDataRow(hasPrn: true, prn: "^%�PRN");
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Data.Tests/Services/DelegateUploadFileServiceTests.cs:261]: This change is a bit surprising

File: Jenkinsfile [hunk 2]
```diff
@@ -71,7 +77,9 @@ pipeline {
             }
             steps {
                 withCredentials([string(credentialsId: 'deploy-test-password', variable: 'PASSWORD')]) {
-                    bat "dotnet publish DigitalLearningSolutions.Web/DigitalLearningSolutions.Web.csproj /p:PublishProfile=DigitalLearningSolutions.Web/Properties/PublishProfiles/PublishToUARTest.pubxml /p:Password=$PASSWORD /p:AllowUntrustedCertificate=True"
+                    nodejs(nodeJSInstallationName: 'NodeJS-16') {
+                        bat "dotnet publish DigitalLearningSolutions.Web/DigitalLearningSolutions.Web.csproj /p:PublishProfile=DigitalLearningSolutions.Web/Properties/PublishProfiles/PublishToTest.pubxml /p:Password=$PASSWORD /p:AllowUntrustedCertificate=True"
+                    }
```
Comment (AlexJacksonDS) [inline Jenkinsfile:82]: This is also the incorrect publish profile for UAR

**Review by AlexJacksonDS — COMMENTED** [review]
I don't see any changes for the DelegateUserDataService, what were the conflicts there?

**Review by DanBloxham-sw — COMMENTED** [review]
I think Alex has spotted the only potential issues with this, the rest looks just about how I'd expect.

I will note that, as part of this, I'm seeing changes to files that I haven't looked at before. I spotted a few style inconsistencies introduced onto files that HEE devs have touched. I'm guessing we haven't got very far with trying to get consistent styling from them?

File: Jenkinsfile (hunk 2 above)
Comment (SteveJacksonSoft) [reply Jenkinsfile:82]: Ohoh, great catch. That was a close one.

Comment (SteveJacksonSoft) [reply Jenkinsfile]: Fixed thisk.

**Pull request merged**